om4labs/diags/moc/moc.py

```diff
@@ -0,0 +1,306 @@
+    msftyyz.units = "kg s-1"
+    msftyyz.coordinates = "region"
+    msftyyz.cell_methods = "z_i:point yq:point time:mean"
+    msftyyz.time_avg_info = "average_T1,average_T2,average_DT"
+    msftyyz.standard_name = "ocean_y_overturning_mass_streamfunction"
+    return msftyyz
+
+
+def moc_maskedarray(vh, mask=None):
+    """Computes the overturning streamfunction given a masked array"""
+    if mask is not None:
+        _mask = np.ma.masked_where(np.not_equal(mask, 1.0), mask)
+    else:
+        _mask = 1.0
+    _vh = vh * _mask
+    _vh_btm = np.ma.expand_dims(_vh[:, -1, :, :] * 0.0, axis=1)
+    _vh = np.ma.concatenate((_vh, _vh_btm), axis=1)
+    _vh = np.ma.sum(_vh, axis=-1) * -1.0
+    _vh = _vh[:, ::-1]  # flip z-axis so running sum is from ocean floor to surface
+    _vh = np.ma.cumsum(_vh, axis=1)
+    _vh = _vh[:, ::-1]  # flip z-axis back to original order
+    return _vh
+
+
+def generate_basin_masks(basin_code, basin=None):
+    """Function to generate pre-defined basin masks"""
+    mask = basin_code * 0
+    if basin == "atlantic_arctic":
+        mask[
+            (basin_code == 2)
+            | (basin_code == 4)
+            | (basin_code == 6)
+            | (basin_code == 7)
+            | (basin_code == 8)
+        ] = 1.0
+    elif basin == "indo_pacific":
+        mask[(basin_code == 3) | (basin_code == 5)] = 1.0
+    elif isinstance(basin, int):
+        mask[(basin_code == basin)] = 1.0
+    else:
+        mask[(basin_code >= 1)] = 1.0
+    return mask
+
+
+def arguments(cliargs=None):
+    """
+    Function to capture the user-specified command line options
+    """
+    description = """ """
+
+    parser = argparse.ArgumentParser(
+        description=description, formatter_class=argparse.RawTextHelpFormatter
+    )
+
+    parser.add_argument(
+        "-b", "--basin", type=str, default=None, help="Path to basin code file"
+    )
+
+    parser.add_argument(
+        "-g", "--gridspec", type=str, default=None, help="Path to gridspec file"
+    )
+
+    parser.add_argument(
+        "infile",
+        metavar="INFILE",
+        type=str,
+        nargs="+",
+        help="Path to input NetCDF file(s)",
+    )
+
+    parser.add_argument(
+        "-i",
+        "--interactive",
+        action="store_true",
+        help="Interactive mode displays plot to screen. Default is False",
+    )
+
+    parser.add_argument(
+        "-l", "--label", type=str, default="", help="String label to annotate the plot"
+    )
+
+    parser.add_argument(
+        "-o",
+        "--outdir",
+        type=str,
+        default="./",
+        help="Output directory. Default is current directory",
+    )
+
+    parser.add_argument(
+        "-f",
+        "--format",
+        type=str,
+        default="png",
+        help="Output format for plots. Default is png",
+    )
+
+    parser.add_argument(
+        "-t", "--topog", type=str, default=None, help="Path to topog file"
+    )
+
+    return parser.parse_args(cliargs)
```

Comment:
rather than returning the parser namespace, I'd return a dict using:

```python
return vars(parser.parse_args(cliargs))
```

Reply:
addressed in dfd833b